addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1611,8 +1608,40 @@ registry.ImagesLazyLoading = publicWidget.Widget.extend({
      * @param {HTMLImageElement} imgEl
      */
     _restoreImage(imgEl) {
-        imgEl.style.minHeight = imgEl.dataset.lazyLoadingInitialMinHeight;
-        delete imgEl.dataset.lazyLoadingInitialMinHeight;
+        this._updateImgMinHeight(imgEl, true);
+    },
+    /**
+     * Updates the image element style with the corresponding min-height.
+     * If the editor is enabled, it deactivates the observer during the CSS
+     * update.
+     *
+     * @param {HTMLElement} imgEl - The image element to update the minimum
+     *        height of.
+     * @param {boolean} [reset=false] - Whether to remove the minimum height
+     *        and restore the initial value.
+     */
+    _updateImgMinHeight(imgEl, reset = false) {
+        if (this.options.wysiwyg) {
+            this.options.wysiwyg.odooEditor.observerUnactive('_updateImgMinHeight');
+        }
+        if (reset) {
+            imgEl.style.minHeight = imgEl.dataset.lazyLoadingInitialMinHeight;
+            if (imgEl.getAttribute('style') === "") {
+                // Don't leave empty `style` attribute on the element. It is
+                // messing with the editor sanitize which removes empty style
+                // and class.
+                imgEl.removeAttribute('style');
+            }
```

Comment:
Not sure I like this :grimacing: If we have to do that in public widgets... then prepare yourself to add many of those :/ I would remove this. Should not it be the sanitize that is fixed instead? Either do that kind of sanitization on already-dirty elements only and/or disable the observer during that kind of sanitization?

Seems louche that a perfectly valid HTML would get dirty just by sanitizing it... would not we COW many things by mistake?

Reply:
Another similar idea: since you re-enable the observer at the end of those operation maybe it could be that function (`observerActive`) which triggers a sanitize of all `style=""` before actually re-enabling the observer.

Reply:
@rdeodoo

Reply:
> Seems louche that a perfectly valid HTML would get dirty just by sanitizing it... would not we COW many things by mistake?

"Many", I don't know, "cowing by mistake", more than probably yes.

I am down to discuss this one tomorrow, it will be easier.

It was not clear in the commit message but note that the real fix is really the part which prevents empty `style` attribute, the observer is really just something I added "just like that" because it's how it's supposed to be.

I am still unsure of what is expected and what is not, but here is what happen (and why writing a word in the `#wrap.oe_structure` ends up sanitizing and marking as o_dirty elements from outside of it: namely the navbar (mitchell admin + logo) and the footer (language selector):
- When you type in the #wrap section to add a letter in the snippet, it will call `_onInput(ev)` from the wysiwyg.
- Which will do `this.historyStep();` > `this.sanitize();` without passing any root/node to `sanitize()`
- As `sanitize` has no target, it will find one with
  https://github.com/odoo/odoo/blob/b9137acb6bb63c3106ccac828edfbef668a02d39/addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js#L802-L815
  The target will be the whole `#wrapwrap` :grimacing: which seems wrong to me, because it's no the current editable area (the #wrap), but the parent, which holds multiple editable area (footer, header, #wrap..)
- Then you get the point, it will sanitize the whole DOM, even unrelated part which have nothing to do with the area where I typed.

I'd like to discuss this one IRL for some guidance.

Reply:
Yep I understood all that, that's why I suggested that it's the sanitization system that should be reviewed :wink: But I of course don't know much about it... editor team... :shrug: We can of course talk about it.